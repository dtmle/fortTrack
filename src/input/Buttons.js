import React from "react";
import "./Button.css";

import { library } from "@fortawesome/fontawesome-svg-core";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faDesktop } from "@fortawesome/free-solid-svg-icons";
import { faXbox, faPlaystation } from "@fortawesome/free-brands-svg-icons";

library.add(faDesktop, faXbox, faPlaystation);

export class InputField extends React.Component {
  render() {
    return (
      <input
        className="InputField"
        placeholder="Display Name"
        value={this.props.value}
        onChange={this.props.onChange}
        style={{
          //change color depending on if valid and changed
          borderColor:
            this.props.valid && this.props.changed
              ? "rgb(0, 255, 0)"
              : "rgb(255, 0, 0)"
        }}
      />
    );
  }
}

export class SendButton extends React.Component {
  render() {
    return (
      <input
        className="SendButton"
        type="submit"
        value={this.props.text}
        onClick={this.props.onClick}
      />
    );
  }
}

export class AddButton extends React.Component {
  render() {
    let progress = "c" + this.props.count;
    return (
      <button
        className={"AddButton " + progress}
        onClick={this.props.onClick}
        style={{ visibility: this.props.hide ? "hidden" : "visible" }}
      >
        +
      </button>
    );
  }
}

export class RemoveButton extends React.Component {
  render() {
    return (
      <button className="RemoveButton" onClick={this.props.onClick}>
        X
      </button>
    );
  }
}

export class RadioGroup extends React.Component {
  render() {
    return (
      <form className="RadioGroup" onChange={this.props.onChange}>
        <p>Platform</p>
        <ul>
          <li className="RadioOption">
            <label>
              <input type="radio" name="platform" value="pc" defaultChecked />
              <span>
                <title>Desktop</title>
                <FontAwesomeIcon icon="desktop" size="2x" />
              </span>
            </label>
          </li>
          <li className="RadioOption">
            <label>
              <input type="radio" name="platform" value="xbl" />
              <span>
                <title>Xbox Live</title>
                <FontAwesomeIcon icon={["fab", "xbox"]} size="2x" />
              </span>
            </label>
          </li>
          <li className="RadioOption">
            <label>
              <input type="radio" name="platform" value="psn" />
              <span>
                <title>Playstation Network</title>
                <FontAwesomeIcon icon={["fab", "playstation"]} size="2x" />
              </span>
            </label>
          </li>
        </ul>
      </form>
    );
  }
}
