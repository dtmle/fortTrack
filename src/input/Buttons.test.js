import React from "react";
import { mount, render, shallow } from "enzyme";
import { InputField, SendButton, AddButton, RemoveButton } from "./Buttons";

describe("InputField", () => {
  it("Should render", () => {
    const component = shallow(<InputField />);
    expect(component).toMatchSnapshot();
  });
  it("Should have a value", () => {
    const component = shallow(<InputField value="Dennis" />);
    expect(component.props().value).toEqual("Dennis");
  });
  it("Should have border red if invalid", () => {
    const component = shallow(<InputField valid={false} />);
    expect(component.props().style.borderColor).toEqual("#A63D40");
  });
  it("Should have border red if unchanged", () => {
    const component = shallow(<InputField changed={false} />);
    expect(component.props().style.borderColor).toEqual("#A63D40");
  });
  it("Should have border green if valid & changed", () => {
    const component = shallow(<InputField valid={true} changed={true} />);
    expect(component.props().style.borderColor).toEqual("#21D3B0");
  });
});

describe("SendButton", () => {
  it("Should render", () => {
    const component = shallow(<SendButton />);
    expect(component).toMatchSnapshot();
  });
  it("Should render text", () => {
    const component = shallow(<SendButton text={"Dennis"} />);
    expect(component.props().value).toEqual("Dennis");
  });
});

describe("AddButton", () => {
  it("Should render", () => {
    const component = shallow(<AddButton />);
    expect(component).toMatchSnapshot();
  });
  it("Should be visible when count < 4", () => {
    const component = shallow(<AddButton hide={false} />);
    expect(component.props().style.visibility).toEqual("visible");
  });
  it("Should be hidden when count == 4", () => {
    const component = shallow(<AddButton hide={true} />);
    expect(component.props().style.visibility).toEqual("hidden");
  });
});

describe("RemoveButton", () => {
  it("Should render", () => {
    const component = shallow(<RemoveButton />);
    expect(component).toMatchSnapshot();
  });
});
