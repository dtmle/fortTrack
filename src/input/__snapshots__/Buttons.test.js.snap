// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddButton Should render 1`] = `
<button
  className="AddButton cundefined"
  style={
    Object {
      "visibility": "visible",
    }
  }
>
  +
</button>
`;

exports[`InputField Should render 1`] = `
<input
  className="InputField"
  maxLength="16"
  placeholder="Name"
  style={
    Object {
      "borderColor": "#A63D40",
    }
  }
/>
`;

exports[`RemoveButton Should render 1`] = `
<button
  className="RemoveButton"
>
  X
</button>
`;

exports[`SendButton Should render 1`] = `
<input
  className="SendButton"
  type="submit"
/>
`;
